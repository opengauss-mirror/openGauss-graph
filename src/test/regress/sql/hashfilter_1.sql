create schema hashfilter_1;
set current_schema = hashfilter_1;

--one-time filter is false in one of branch of setop
START TRANSACTION;
create table t1 (a int, d int);
create table t2 (a int8, d int);
create view v1 as select * from t1 union all select * from t1;
create view v2 as select * from t2 union all select * from t2;
create view v as 
select * from v1 where d > 6 and d < 9
union all
select * from v2 where d > 9 and d < 11;

DROP table t1 cascade;
DROP table t2 cascade;

create table t1 (col_int integer,col_numeric numeric,col_varchar varchar,col_date date) with(orientation=column);
create table t2 (col_int integer,col_numeric numeric,col_varchar varchar,col_date date) with(orientation=column);
insert into t1 values(1,1,'aaa','2010-10-01');
insert into t1 values(2,2,'bbb','2010-11-01');
insert into t1 values(3,3,'ccc','2010-12-01');
insert into t2 select * from t1;

--hashfilter expr select other qual
explain (costs off) select t1.* from t1 right join t2 on t2.col_int = trunc(t1.col_numeric) and t1.col_varchar like 'a%' where t2.col_date is not null and t2.col_int < 750 order by t1.col_int;
select t1.* from t1 right join t2 on t2.col_int = trunc(t1.col_numeric) and t1.col_varchar like 'a%' where t2.col_date is not null and t2.col_int < 750 order by t1.col_int;

COMMIT;

--drop schema
reset plan_mode_seed;

--hashfilter for muti-nodegroup


-- clean up
reset current_schema;
drop schema hashfilter_1 cascade;
