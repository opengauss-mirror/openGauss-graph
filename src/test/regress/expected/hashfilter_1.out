create schema hashfilter_1;
set current_schema = hashfilter_1;
--one-time filter is false in one of branch of setop
START TRANSACTION;
create table t1 (a int, d int);
create table t2 (a int8, d int);
create view v1 as select * from t1 union all select * from t1;
create view v2 as select * from t2 union all select * from t2;
create view v as 
select * from v1 where d > 6 and d < 9
union all
select * from v2 where d > 9 and d < 11;
DROP table t1 cascade;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to view v1
drop cascades to view v
DROP table t2 cascade;
NOTICE:  drop cascades to view v2
create table t1 (col_int integer,col_numeric numeric,col_varchar varchar,col_date date) with(orientation=column);
create table t2 (col_int integer,col_numeric numeric,col_varchar varchar,col_date date) with(orientation=column);
insert into t1 values(1,1,'aaa','2010-10-01');
insert into t1 values(2,2,'bbb','2010-11-01');
insert into t1 values(3,3,'ccc','2010-12-01');
insert into t2 select * from t1;
--hashfilter expr select other qual
explain (costs off) select t1.* from t1 right join t2 on t2.col_int = trunc(t1.col_numeric) and t1.col_varchar like 'a%' where t2.col_date is not null and t2.col_int < 750 order by t1.col_int;
                                 QUERY PLAN                                  
-----------------------------------------------------------------------------
 Row Adapter
   ->  Vector Sort
         Sort Key: t1.col_int
         ->  Vector Hash Left Join
               Hash Cond: ((t2.col_int)::numeric = trunc(t1.col_numeric, 0))
               ->  CStore Scan on t2
                     Filter: ((col_date IS NOT NULL) AND (col_int < 750))
               ->  CStore Scan on t1
                     Filter: ((col_varchar)::text ~~ 'a%'::text)
(9 rows)

select t1.* from t1 right join t2 on t2.col_int = trunc(t1.col_numeric) and t1.col_varchar like 'a%' where t2.col_date is not null and t2.col_int < 750 order by t1.col_int;
 col_int | col_numeric | col_varchar |         col_date         
---------+-------------+-------------+--------------------------
       1 |           1 | aaa         | Fri Oct 01 00:00:00 2010
         |             |             | 
         |             |             | 
(3 rows)

COMMIT;
--drop schema
reset plan_mode_seed;
--hashfilter for muti-nodegroup
-- clean up
reset current_schema;
drop schema hashfilter_1 cascade;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to table hashfilter_1.t1
drop cascades to table hashfilter_1.t2
